### ISSUE TITLE
Neovim 0.8.0's built-in terminal is broken on older versions of Windows 10

### ISSUE THREAD

**Issue by Letheward** [neovim/neovim#20458]
### Neovim version (nvim -v)

NVIM v0.8.0-1210-gd367ed9b2

### Vim (not Nvim) behaves the same?

No, Vim 8.2.1895

### Operating system/version

Windows 10 1703

### Terminal name/version

nvim-qt

### $TERM environment variable

none

### Installation

Github Releases, manual installation

### How to reproduce the issue

1. Download Neovim
2. Open it using `nvim-qt`
3. Try to open the built-in terminal:
~~~
:term
~~~

---

To reproduce using `nvim --clean` I need to start from `cmd.exe`:

1. Download Neovim

2. Open `cmd.exe`, and go into the `nvim-win64/bin` folder.

3. Open Neovim using `--clean` flag:
~~~
nvim --clean
~~~

4. Try to open the built-in terminal:
~~~
:term
~~~

### Expected behavior

Neovim opens the built-in terminal and starts the `cmd` shell in there.

### Actual behavior

Neovim didn't open the built-in terminal, and displayed this error:
~~~
E903: Process failed to start: function not implemented: "C:\Windows\system32\cmd.exe"
~~~

**Comment by zeertzjq** [comment]
Duplicate of #20420

**Comment by justinmk** [comment]
Not sure how `:term` could fail because we run it in CI.

Anyways, can you try the development version @Letheward ? I am wondering if https://github.com/neovim/neovim/pull/19438 helps anything here.

The [Releases](https://github.com/neovim/neovim/releases/tag/nightly) page has pre-built archives Windows.

**Comment by Letheward** [comment]
> Anyways, can you try the development version @Letheward ? I am wondering if #19438 helps anything here. The [Releases](https://github.com/neovim/neovim/releases/tag/nightly) page has a pre-built zip for Windows.

I tried `NVIM v0.9.0-dev-16-g56998feeb`, it still showed the same error.

**Comment by Letheward** [comment]
Not sure if this is related, but Windows ConPTY is [introduced](https://devblogs.microsoft.com/commandline/windows-command-line-introducing-the-windows-pseudo-console-conpty/) in Windows 10 1809, which is newer than my version (1703). 

Some terminal emulators (like [alacritty](https://github.com/alacritty/alacritty)) use ConPTY, and indeed I can't use them on older versions of Windows 10.

**Comment by zeertzjq** [comment]
This issue seems to contain more information with #20420

**Comment by bfredl** [comment]
can we feature detect and give a more informative error message than ` function not implemented: "C:\Windows\system32\cmd.exe"` perhaps?

**Comment by erw7** [comment]
> can we feature detect and give a more informative error message than function not implemented: "C:\Windows\system32\cmd.exe" perhaps?

Yes. I think the error message should be improved. However, support for Version 1803 and earlier for the general public ended a long time ago. Therefore, it is not a priority.

Ref. https://learn.microsoft.com/en-us/lifecycle/products/windows-10-home-and-pro

**Comment by justinmk** [comment]
I'm seeing this on a Windows 2016 VM. Running `winver` in cmd.exe shows `Version 1607`.

From https://learn.microsoft.com/en-us/lifecycle/products/windows-10-home-and-pro I see that 1607 "ended" in 2018:

```
Version        Start Date       End Date
-----------------------------------------
Version 1607 | Aug 2, 2016 | Apr 10, 2018
```

those 2-year cycles are rather aggressive.
This Windows VM is just the standard AMI used by https://aws.amazon.com/workspaces/

So in general, those 2-year cycles don't decide when we drop support for a Windows version. But in this case for `:terminal`, we explicitly require Version 1809 or newer.

**Comment by ymic9963** [comment]
For this issue would an emitting an error in Neovim be the best option or would a note in the docs suffice? We can get the version string with 
```
systeminfo | findstr /B /C:"OS Version"
``` 
in `cmd.exe`. Then it's just a matter of parsing it and checking if it's the correct version number. I'm not really familiar with the codebase so implementing this seems pretty daunting tbh. Hopefully my snippet here helps push it to get it closed.

**Comment by justinmk** [comment]
> For this issue would an emitting an error in Neovim be the best option

Detection is always better if possible. Is something missing from the docs currently? `:help support`

> Then it's just a matter of parsing it

Try `vim.version.parse()` on the output.

**Comment by ymic9963** [comment]
A typical version string using that looks like,
```
OS Version:                    10.0.26100 N/A Build 26100
```
and using `vim.version.parse("OS Version:                    10.0.26100 N/A Build 26100")` does work...I just don't know where to use it...

**Comment by justinmk** [comment]
https://github.com/neovim/neovim/blob/5370b7a2e0a0484c9005cb5a727dffa5ef13b1ed/src/nvim/ex_docmd.c#L7970

```c
#ifdef MSWIN
  Array a = ARRAY_DICT_INIT;
  Error err = ERROR_INIT;
  Object rv = NLUA_EXEC_STATIC("return ...", a, kRetObject, NULL, &err);
  api_clear_error(&err);
  if (!rv.data.boolean) {
    ...
  }
#endif
```

Or perhaps we could set a flag during startup. And then show the error if `:terminal` is attempted.
